.github/workflows/version-or-publish.yml

```diff
@@ -0,0 +1,30 @@
+name: Version or Publish
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
we should be able to rename the base branch now - we need to preserve `master` as artifact of the past though as all existing users are using that ref at the moment